simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/ConfirmationFragment.kt

```diff
@@ -0,0 +1,175 @@
+import android.os.Bundle
+import android.os.Environment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import org.nypl.simplified.cardcreator.R
+import org.nypl.simplified.cardcreator.databinding.FragmentConfirmationBinding
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.io.FileOutputStream
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.Date
+
+class ConfirmationFragment : Fragment() {
+
+  private val logger = LoggerFactory.getLogger(ConfirmationFragment::class.java)
+
+  private var _binding: FragmentConfirmationBinding? = null
+  private val binding get() = _binding!!
+
+  private lateinit var type: String
+  private lateinit var username: String
+  private lateinit var barcode: String
+  private lateinit var pin: String
+  private var temporary: Boolean = true
+  private lateinit var message: String
+
+  private val storageRequestCode = 203
+  private val dateFormat = "dd-MM-yyyy"
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    _binding = FragmentConfirmationBinding.inflate(inflater, container, false)
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    arguments?.let {
+      val bundle = ConfirmationFragmentArgs.fromBundle(it)
+      type = bundle.type
+      username = bundle.username
+      barcode = bundle.barcode
+      pin = bundle.pin
+      temporary = bundle.temporary
+      message = bundle.message
+
+      binding.nameCard.text = bundle.name
+      binding.cardBarcode.text = getString(R.string.user_card_number, bundle.barcode)
+      binding.cardPin.text = getString(R.string.user_pin, bundle.pin)
+      binding.headerStatusDescTv.text = bundle.message
+
+      val currentDate: String = SimpleDateFormat(dateFormat, Locale.getDefault()).format(Date())
+      binding.issued.text = getString(R.string.issued_date, currentDate)
+    }
+
+    // Go to next screen
+    binding.nextBtn.setOnClickListener {
+      logger.debug("User created card")
+      returnResult()
+    }
+
+    // Go to previous screen
+    binding.prevBtn.setOnClickListener {
+
+      if (ContextCompat.checkSelfPermission(activity!!, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        != PackageManager.PERMISSION_GRANTED
+      ) {
+        logger.debug("Requesting storage permission")
+        ActivityCompat.requestPermissions(
+          activity!!,
+          arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+          storageRequestCode
+        )
+      } else {
+        createDigitalCard()
+      }
+    }
+  }
+
+  /**
+   * Create library card from ImageView
+   */
+  private fun createDigitalCard() {
```

Comment:
Might be worth doing this work on a background thread using a coroutine.

Reply:
Oh yea! I will add that.